lisa/lisa-core/src/main/java/it/unive/lisa/analysis/string/CharInclusion.java

```diff
@@ -0,0 +1,297 @@
+		if (this.getCertainlyContained().size() > other.getCertainlyContained().size() ||
+				other.getMaybeContained().size() > this.getMaybeContained().size())
+			return false;
+
+		if (!other.getCertainlyContained().containsAll(this.getCertainlyContained()))
+			return false;
+
+		return this.getMaybeContained().containsAll(other.getMaybeContained());
+	}
+
+	@Override
+	public boolean equals(Object o) {
+		if (this == o)
+			return true;
+		if (o == null || getClass() != o.getClass())
+			return false;
+		CharInclusion that = (CharInclusion) o;
+		return Objects.equals(certainlyContained, that.certainlyContained)
+				&& Objects.equals(maybeContained, that.maybeContained);
+	}
+
+	@Override
+	public int hashCode() {
+		return Objects.hash(certainlyContained, maybeContained);
+	}
+
+	@Override
+	public CharInclusion top() {
+		return TOP;
+	}
+
+	@Override
+	public CharInclusion bottom() {
+		return BOTTOM;
+	}
+
+	@Override
+	public boolean isTop() {
+		return this.equals(TOP);
+	}
+
+	@Override
+	public boolean isBottom() {
+		return this.equals(BOTTOM);
+	}
+
+	@Override
+	public DomainRepresentation representation() {
+		if (isBottom())
+			return Lattice.bottomRepresentation();
+		if (isTop())
+			return Lattice.topRepresentation();
+
+		return new StringRepresentation(formatRepresentation());
+	}
+
+	/**
+	 * Yields the set of certainly contained characters of this abstract value.
+	 *
+	 * @return the set of certainly contained characters of this abstract value.
+	 */
+	public Collection<Character> getCertainlyContained() {
+		return this.certainlyContained;
+	}
+
+	/**
+	 * Yields the set of maybe contained characters of this abstract value.
+	 *
+	 * @return the set of maybe contained characters of this abstract value.
+	 */
+	public Collection<Character> getMaybeContained() {
+		return this.maybeContained;
+	}
+
+	private String formatRepresentation() {
+		StringBuilder stringBuilder = new StringBuilder("CertainlyContained: {");
+		int counter = 0;
+
+		for (Character certainlyContainedChar : this.getCertainlyContained()) {
+			String formattedCharacter;
+
+			formattedCharacter = counter != this.getCertainlyContained().size() - 1 ? certainlyContainedChar + ", "
+					: certainlyContainedChar + "}";
+			counter++;
+
+			stringBuilder.append(formattedCharacter);
+		}
+
+		if (counter == 0) {
+			stringBuilder.append("}");
+		}
+
+		counter = 0;
+		stringBuilder.append(", MaybeContained: {");
+
+		for (Character maybeContainedChar : this.getMaybeContained()) {
+			String formattedCharacter;
+
+			formattedCharacter = counter != this.getMaybeContained().size() - 1 ? maybeContainedChar + ", "
+					: maybeContainedChar + "}";
+			counter++;
+
+			stringBuilder.append(formattedCharacter);
+		}
+
+		if (counter == 0) {
+			stringBuilder.append("}");
+		}
+
+		return stringBuilder.toString();
+	}
+
+	@Override
+	public CharInclusion evalNullConstant(ProgramPoint pp) {
```

Comment:
This is the default behavior, you can remove this for clarity (valid also for the other `return TOP` and `return Satisfiability.UNKNOWN` below)